src/OrchardCore.Modules/OrchardCore.Lucene/LuceneIndexManager.cs

```diff
@@ -246,7 +245,10 @@ private BaseDirectory CreateDirectory(string indexName)
                     path.Create();
                 }
 
-                return FSDirectory.Open(path);
+                lock (_synLock)
```

Comment:
Explain, because I think it's wrong, in the sense that this instance is a singleton, and the whole method is already locked on this instance, so we can only need a `lock (_synlock)` if two tenants access the same path, which would mean two instances deleting the same lucene folder, which is NOT supported ;)

Reply:
Forget about this, I just read your comment